cli/azd/pkg/project/service_target_aks.go

```diff
@@ -606,6 +606,7 @@ func (t *aksTarget) ensureClusterContext(
 		convertOptions := &kubelogin.ConvertOptions{
 			Login:      "azd",
 			KubeConfig: kubeConfigPath,
+			TenantId:   t.env.GetTenantId(),
```

Comment:
tenantId is not ensured to be in the azd-environment.

We have a way to resolve the tenantId for a project here: https://github.com/Azure/azure-dev/blob/main/cli/azd/cmd/auth_token.go#L84

Most cases, it is resolved from the subscription Id, which is ensured to be in the azd-env

Reply:
Let me ping @wbreza as I am not very familiar with the `managedCluster`